api/types/types.go

```diff
@@ -0,0 +1,32 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package types
+
+// ModelType indicates a model type.
+type ModelType string
+
+const (
+	// IAAS is the type for IAAS models.
+	IAAS ModelType = "iaas"
+
+	// CAAS is the type for CAAS models.
+	CAAS ModelType = "caas"
```

Comment:
Should we introduce a test to make sure these values haven't drifted from core? I know it probably isn't needed but I like things to go bang if something changes.